sale_cancel_picking/sale.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#  License AGPL version 3 or later
+#  See license in __openerp__.py or http://www.gnu.org/licenses/agpl-3.0.txt
+#  Copyright (C) 2014 Akretion (http://www.akretion.com).
+#  @author Adrien CHAUSSENDE <adrien.chaussende@akretion.com>
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+
+
+class SaleOrder(orm.Model):
+    _inherit = 'sale.order'
+
+    _columns = {
+        'cancel_log': fields.html("Cancellation Logs"),
+    }
+
+    def _cancel_linked_record(self, cr, uid, order, context=None):
+        """
+        Method that cancels the pickings related to the order. It writes
+        messages on the cancellation logs attribute.
+        """
+        for picking in order.picking_ids:
+            able_to_cancel, message, important = \
+                picking.can_cancel_picking_out()
+            if able_to_cancel:
+                picking.action_cancel()
+            order.add_cancel_log(message, important=important)
+
+    def action_cancel(self, cr, uid, ids, context=None):
+        for order in self.browse(cr, uid, ids, context=context):
+            self._cancel_linked_record(cr, uid, order, context=None)
```

Comment:
I think it would be more clear if you make _cancel_linked_record() a standard orm method and use the line order._cancel_linked_record().